src/modules/admin/navigations/controllers/navigations.controller.ts

```diff
@@ -0,0 +1,77 @@
+import {
+	Body,
+	Controller,
+	Delete,
+	Get,
+	Logger,
+	Param,
+	Post,
+	Put,
+	Query,
+	Session,
+} from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
+import { IPagination } from '@studiohyperdrive/pagination/dist/lib/pagination.types';
+
+import { CreateNavigationDto, NavigationsQueryDto } from '../dto/navigations.dto';
+import { NavigationsService } from '../services/navigations.service';
+import { Navigation } from '../types';
+
+import { DeleteResponse } from '~shared/types/types';
+
+@ApiTags('Navigations')
+@Controller('admin/navigations')
+export class NavigationsController {
+	private logger: Logger = new Logger(NavigationsController.name, { timestamp: true });
+
+	constructor(private navigationsService: NavigationsService) {}
+
+	@Get()
+	public async getNavigations(
+		@Query() navigationsQueryDto: NavigationsQueryDto
+	): Promise<IPagination<Navigation>> {
+		const navigations = await this.navigationsService.findAll(navigationsQueryDto);
+		return navigations;
+	}
+
+	/**
+	 * Get navigation items for the current user (logged in or not logged in)
+	 * currently there are not yet special permission groups
+	 */
+	@Get('items')
+	async getNavigationItems(
+		@Session() session: Record<string, any>
+	): Promise<Record<string, Navigation[]>> {
+		const navigations = await this.navigationsService.getNavigationItems(session);
+		return navigations;
```

Comment:
we could remove the "user groups" property from these objects, since it isn't useful for the frontend. But on the other had, it doesn't really matter, since user groups are no secret.